### PR TITLE
vmdktool 1.4 (new formula)

### PR DESCRIPTION
**Pull request by glennmatthews**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/vmdktool.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Vmdktool < Formula
+  desc "Converts raw filesystems to VMDK files and vice versa"
+  homepage "http://manned.org/vmdktool"
+  url "https://people.freebsd.org/~brian/vmdktool/vmdktool-1.4.tar.gz"
+  sha256 "981eb43d3db172144f2344886040424ef525e15c85f84023a7502b238aa7b89c"
+
+  def install
+    system "make", "CFLAGS='-D_GNU_SOURCE -g -O -pipe'"
+    bin.mkpath
```
Comment (woodruffw) [inline Formula/vmdktool.rb]: Don't worry about these `mkpath`s. Homebrew will handle them.

File: Formula/vmdktool.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    man8.mkpath
+    # vmdktool makefile doesn't recognize --manpath option, so force it:
+    inreplace "Makefile", "man/man8", "share/man/man8"
```
Comment (woodruffw) [inline Formula/vmdktool.rb]: Has upstream been alerted of this?

**Review by woodruffw — CHANGES_REQUESTED** [review]

File: Formula/vmdktool.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    system "make", "install", "PREFIX=#{prefix}"
+  end
+
+  test do
+    # Yes, it's a bad test, but vmdktool only does conversions of existing disk images
```
Comment (ilovezfs) [inline Formula/vmdktool.rb]: You can do this
```
    system "dd", "if=/dev/zero", "of=raw", "bs=512", "count=20480"
    system "#{bin}/vmdktool", "-v", "vd", "raw"
    assert_match "RDONLY 20480 SPARSE", pipe_output("strings", File.read("vd"))
```

Comment (ilovezfs) [inline Formula/vmdktool.rb]: or even just
```
    system "dd", "if=/dev/zero", "of=raw", "bs=512", "count=20480"
    system "#{bin}/vmdktool", "-v", "vd", "raw"
    assert_match "RDONLY 20480 SPARSE", shell_output("#{bin}/vmdktool -i vd")
```

File: Formula/vmdktool.rb (hunk 2 above)
Comment (glennmatthews) [inline Formula/vmdktool.rb]: No, but will do. Thanks.

File: Formula/vmdktool.rb (hunk 3 above)
Comment (glennmatthews) [inline Formula/vmdktool.rb]: Great suggestion! For my use cases I'm not generally creating new images, but this is a great way to validate this. Will do.

File: Formula/vmdktool.rb (hunk 1 above)
Comment (glennmatthews) [inline Formula/vmdktool.rb]: Hm, what am I missing? If I remove these, I get:

```
==> make CFLAGS='-D_GNU_SOURCE -g -O -pipe'
==> make install PREFIX=/usr/local/Cellar/vmdktool/1.4
Last 15 lines from /Users/glmatthe/Library/Logs/Homebrew/vmdktool/02.make:
2017-01-23 11:22:09 -0500

make
install
PREFIX=/usr/local/Cellar/vmdktool/1.4

install -s vmdktool /usr/local/Cellar/vmdktool/1.4/bin/
install: /usr/local/Cellar/vmdktool/1.4/bin/: No such file or directory
make: *** [install] Error 71
```

Comment (woodruffw) [inline Formula/vmdktool.rb]: Sorry, ignore me. I didn't realize this was prefixed inside the cellar; you'll need those `mkpath`s after all.

File: Formula/vmdktool.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/vmdktool.rb]: @glennmatthews After creating an issue please add a link to the upstream issue and explanation of why the patch is needed in a comment in the formula file. Also, please do this for the `mkpath`s above too. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by glennmatthews** [comment]
Hi @MikeMcQuaid - I'm not aware of any public issue tracker for vmdktool. For past bug reports I've emailed the maintainer directly (brian@awfulhak.org) and he's been quite responsive. I emailed him about the MANDIR issue yesterday but haven't seen a response yet. 

Is the issue with the `mkpath`s that the Makefile isn't doing the appropriate `mkdir`s when doing a `make install`? Or what other fix would be desired here?

**Comment by MikeMcQuaid** [comment]
> Hi @MikeMcQuaid - I'm not aware of any public issue tracker for vmdktool. For past bug reports I've emailed the maintainer directly (brian@awfulhak.org) and he's been quite responsive. I emailed him about the MANDIR issue yesterday but haven't seen a response yet.

That's fine, in that case we just want to note what date they were emailed and a summary of the email and/or their response.

> Is the issue with the mkpaths that the Makefile isn't doing the appropriate mkdirs when doing a make install? Or what other fix would be desired here?

Indeed, do `mkdir` or ideally `mkdir -p` before assuming a directly exists.

**Comment by glennmatthews** [comment]
Updated diffs with improved test as requested by @ilovezfs and comment about contact with upstream maintainer as requested by @MikeMcQuaid. Thanks!

**Review by woodruffw — APPROVED** [review]

**Comment by glennmatthews** [comment]
@MikeMcQuaid any remaining concerns I need to address?

**Review by MikeMcQuaid — APPROVED** [review]

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**